Video/tormyvancool_YOUTUBE Downloader.lua

```diff
@@ -0,0 +1,367 @@
+--   [linux] yt-dlp/yt-dlp_linux https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp
+--   [darwin] yt-dlp/yt-dlp_macos https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp_macos
+-- @donation Pay me a drink https://bit.ly/Pay_Me_a_Drink
+-- @about
+--   # Import VIDEOs directly in TimeLine from YouTUBE, VIMEO, PATREONS and thousand other ones.
+--  
+--       Key Features:
+-- 
+--       - 4 click operation: Start the script, enter the URL, give a title, click on OK
+--       - Import any Video in TimeLine by giving just the URL
+--       - Videos are saved into the project folder under the dedicated /Videos/ folder
+--       - Videos are imported into a new track, having the given name, and at the cursor position
+--       - Auto-update of the binaries "yt-dlp" each time the script is invoked, to ensure top quality at each use
+--       - Compatible with about thousand platforms included:
+--       - YouTUBE
+--       - Vimeo
+--       - Patreons
+--       and several other ones ...
+-- 
+--      [Full list here](https://github.com/yt-dlp/yt-dlp/blob/master/supportedsites.md)
+--
+--   #Credits:
+--       Stefano marcantoni and Talagan - to have helped for MAC implementation
+--       Paolo Saggese - to have helped for Linux implementation
+
+reaper.ClearConsole()
+
+---------------------------------------------
+-- MAIN VARIABLES
+---------------------------------------------
+local LF = "\n"
+local pipe = "|"
+local colon = ":"
+local quote = '"'
+local slash = '\\'
+local backslash = '/'
+local clock = os.clock
+local debug = false
+local ver = 2.9
+local InputVariable = ""
+local dlpWin = 'yt-dlp.exe'
+local dlpMac = 'yt-dlp_macos'
+local dlpLnx = 'yt-dlp_linux'
+local version = reaper.GetAppVersion()
+local pj_name_ = reaper.GetProjectName(0, "")
+local ProjDir = reaper.GetProjectPathEx(0)
+local ResourcePATH = reaper.GetResourcePath()
+local VideoPath = 'Video'
+local CallPath = ResourcePATH .. '/Scripts/Tormy Van Cool ReaPack Scripts/' .. VideoPath .. '/yt-dlp/' -- Get FullPath to yt-dlp
+
+---------------------------------------------
+-- FUNCTIONS
+---------------------------------------------
+      
+      -- SLEEP(SECONDS)
+      function sleep(n)
+        local t0 = clock()
+        while clock() - t0 <= n do end
+      end
+      
+      -- IDENTIFIES THE O.S.
+      function getOS()
+        local OS = reaper.GetOS()
+        local a = {}
+        local MainPath = ''
+        if OS == "Win32" or OS == "Win64" then
+          MainPath = '"' .. ResourcePATH .. '/Scripts/Tormy Van Cool ReaPack Scripts/' .. VideoPath .. '/yt-dlp/' .. dlpWin .. '"'
+          Start = 'start /b /wait "UPDATE & DOWNLOAD" '
+          OpSys = 1
+        end
+        if OS == "OSX32" or OS == "OSX64" or OS == "macOS-arm64" then
+          MainPath  = './yt-dlp_macos'
+          Start = 'cd "' .. CallPath .. '" && chmod +x ' .. dlpMac .. ' && '
+          os.execute('chmod +x "' ..  MainPath .. '"')
```

Comment:
Same issue as last time here.

Reply:
**If you are talking about the line 145**: it seems in Mac when you call directly the application (1 single command instead of all that "CDs") it returns that the file doesn't exists.

**If you are talking about line146**: On Linux I tried to give more than one time `chmod +x` everything works. So I don't see evidence to do it in a different way since when the yt-dlp is downloaded it seems do not have the execution flags. Only readable.

Reply:
Line 146 only runs on macOS. It attempts to set the execute bit on yt-dlp_macos in whichever directory is current (unlikely to be yours the first time). So the line is effectively no-op. Only the one at line 145 is effective (ignoring the same injection problem). By the way it's checking `OS == "OSX32"` despite the script (well, the yt-dlp binary) not supporting macOS 32-bit?

Reply:
> Line 146 only runs on macOS. It attempts to set the execute bit on yt-dlp_macos in whichever directory is current (unlikely to be yours the first time). So the line is effectively no-op. Only the one at line 145 is effective (ignoring the same injection problem). By the way it's checking `OS == "OSX32"` despite the script (well, the yt-dlp binary) not supporting macOS 32-bit?

Actually even the very first time is my directory.
It creates firs the yt-dlp subtir
then, on it, it will download the yt-dlp.something
Only from that moment on the flag is changed on +x

in can't be otherwise. or it means the script is not executed linearly form line 1 to the last and something si skipped.
But if so, it means something else is not working as it should.

NOTE: I solved the checking for OSX2 that however it didn't do anything wrong if there.
I have always the same if then else end on a separate file and I copy and paste it.
If the script is installed on any machine: ti works.
And that check is not damaging anything.

Reply:
> It creates firs the yt-dlp subtir
> then, on it, it will download the yt-dlp.something
>
> in can't be otherwise

That's not in the script's code? `getOS` is called from line 175 and it's the first thing that's done after clearing the console and defining a few variables and functions. The directory only gets changed at line 312 when a command built from `Start` gets executed?

Reply:
> > It creates firs the yt-dlp subtir
> > then, on it, it will download the yt-dlp.something
> > in can't be otherwise
> 
> That's not in the script's code? `getOS` is called from line 175 and it's the first thing that's done after clearing the console and defining a few variables and functions. Nothing changes the current working directory of REAPER's process.
> 
> The `yt-dlp` directory is created and filled when the package is installed by ReaPack. Not when running the script. And ReaPack does so without messing with REAPER's working dir.

yes ReaPack does it because the code is telling to do so, due the lines @provide.
Hence the `yt-dlp` subdir is created, and the script behaves in the way I said.

Reply:
I need your feedback to work on this.

Reply:
Committed

Reply:
I reopened this conversation because I see I can perhaps review that function that selects the OSes. I will try to make it better. Give me some time. I will eb back when ready by closing this conversation and update the code

Reply:
Ok super. Committed. the path is detected by the Reaper's APIs and it's not more hardencoded

Reply:
> EDIT: instead if you want the script goes to check if really that subdir

I didn't say that? There is no reason to check whether ReaPack has installed yt-dlp. It either does or the entire package is not installed at all & the user is informed of the error.

I was talking about the broken & redundant chmod which is still not removed.